endaq/batch/core.py

```diff
@@ -366,7 +381,15 @@ def add_pvss(self, init_freq: float = 1.0, bins_per_octave: float = 3.0):
         :param init_freq: the first frequency sample in the spectrum
         :param bins_per_octave: the number of samples per frequency octave
         """
-        self._metrics_queue["pvss"] = _make_pvss
+        if (
+            self._pvss_init_freq is not None
+            # `self._pvss_init_freq` may be set by `add_metrics`
+            # -> check metrics queue entries specifically
+            and any(name == "pvss" for (name, _) in self._metrics_queue)
+        ):
```

Comment:
From a readability angle, comments between the clauses of a conditional are probably a bad idea. It reads the same with the comments before the code. Personally, I'd put the comments before the `if`, but that's just me.

Reply:
addressed in 2bcc6590a792f56ba56af799353e24205d686e1d